ToolBox/SmartInsider/SmartInsiderProgram.cs

```diff
@@ -23,9 +23,19 @@ public static class SmartInsiderProgram
         /// <summary>
         /// Converts Smart Insider data
         /// </summary>
-        public static void SmartInsiderConverter(DateTime date, string sourceDirectory, string destinationDirectory)
+        public static void SmartInsiderConverter(DateTime date, string sourceDirectory, string destinationDirectory, string processedFileDirectory = null)
         {
-            var converter = new SmartInsiderConverter(new DirectoryInfo(sourceDirectory), new DirectoryInfo(destinationDirectory));
+            // Default to the Data Folder Smart Insider data directory if no argument is passed
+            if (string.IsNullOrEmpty(processedFileDirectory))
+            {
+                processedFileDirectory = Path.Combine(Globals.DataFolder, "alternative", "smartinsider");
```

Comment:
Isn't this default value already handled by the caller using `GetParameterOrDefault()`?

Reply:
We need to set a default for external callers of this method